modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js

```diff
@@ -67,84 +64,68 @@ export default class Tile3DLayer extends CompositeLayer {
     return info;
   }
 
-  async _loadTileset(tilesetUrl, fetchOptions, ionMetadata) {
-    const response = await fetch(tilesetUrl, fetchOptions);
-    const tilesetJson = await response.json();
-
-    const loadOptions = this.getLoadOptions();
-
-    const tileset3d = new Tileset3D(tilesetJson, tilesetUrl, {
-      onTileLoad: tileHeader => {
-        this.props.onTileLoad(tileHeader);
-        this._updateTileset(tileset3d);
-      },
+  async _loadTileset(tilesetUrl) {
+    const {loader, loadOptions} = this.props;
+    const tilesetJson = await load(tilesetUrl, loader, loadOptions);
+    const tileset3d = new Tileset3D(tilesetJson, {
+      onTileLoad: this._onTileLoad.bind(this),
       onTileUnload: this.props.onTileUnload,
       onTileLoadFail: this.props.onTileError,
-      // TODO: explicit passing should not be needed, registerLoaders should suffice
-      fetchOptions,
-      ...ionMetadata,
-      ...loadOptions
+      fetchOptions: loadOptions
     });
 
     this.setState({
       tileset3d,
       layerMap: {}
     });
 
-    if (tileset3d) {
-      this._updateTileset(tileset3d);
-      this.props.onTilesetLoad(tileset3d);
-    }
+    this._updateTileset(tileset3d);
+    this.props.onTilesetLoad(tileset3d);
   }
 
-  async _loadTilesetFromIon(ionAccessToken, ionAssetId) {
-    const ionMetadata = await _getIonTilesetMetadata(ionAccessToken, ionAssetId);
-    const {url, headers} = ionMetadata;
-    await this._loadTileset(url, {headers}, ionMetadata);
+  _onTileLoad(tileHeader) {
+    this.props.onTileLoad(tileHeader);
+    this._updateTileset(this.state.tileset3d);
+    this.setNeedsUpdate();
```

Comment:
Why is this needed? What happened to `frameNumber`?

Reply:
> Why is this needed?
After `this._updateTileset`, the `renderLayers` should be called, which it isn't if I don't manually call `this.setNeedsUpdate()`.

> What happened to frameNumber?
`frameNumber` is not needed to decide if the tile should be selected for rendering, `tile.selected` is enough.

Reply:
`frameNumber` is used to inform the layer whether the selected tiles have changed. We should not have to rerender on every viewport change.